identity_credential/src/credential/did_configuration_resource.rs

```diff
@@ -0,0 +1,302 @@
+      issuance_date: self.issuance_date.unwrap_or_else(Timestamp::now_utc),
+      expiration_date: Some(self.expiration_date.ok_or(Error::MissingExpirationDate)?),
+      credential_status: None,
+      credential_schema: Vec::new().into(),
+      refresh_service: Vec::new().into(),
+      terms_of_use: Vec::new().into(),
+      evidence: Vec::new().into(),
+      non_transferable: None,
+      properties: Object::new(),
+      proof: None,
+    })
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use crate::credential::did_configuration_resource::DIDConfigurationResource;
+  use crate::credential::did_configuration_resource::DomainLinkageCredentialBuilder;
+  use crate::credential::Credential;
+  use crate::credential::Issuer;
+  use crate::error::Result;
+  use crate::Error;
+  use identity_core::common::{Object, OrderedSet};
+  use identity_core::common::Timestamp;
+  use identity_core::common::Url;
+  use identity_core::convert::FromJson;
+  use identity_did::did::CoreDID;
+  use identity_did::did::CoreDIDUrl;
+  use identity_did::service::Service;
+  use serde_json::json;
+
+  #[test]
+  fn builder() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Credential = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build()
+      .unwrap();
+    let _config: DIDConfigurationResource = DIDConfigurationResource::new(vec![credential]);
+  }
+
+  #[test]
+  fn builder_no_issuer() {
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingIssuer)));
+  }
+
+  #[test]
+  fn builder_no_origin() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingOrigin)));
+  }
+
+  #[test]
+  fn builder_no_expiration_date() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingExpirationDate)));
+  }
+
+  #[test]
+  fn from_json() {
+    const JSON1: &str = include_str!("../../tests/fixtures/well-known-configuration.json");
+    let _well_known: DIDConfigurationResource = DIDConfigurationResource::from_json(JSON1).unwrap();
```

Comment:
Nit: In these cases it's fine to:
```suggestion
    DIDConfigurationResource::from_json(JSON1).unwrap();
```